src/web/client/utilities/urlBuilderUtil.ts

```diff
@@ -216,6 +217,34 @@ export function getLogicalEntityName(result: any, logicalEntityName?: string) {
     return logicalEntity;
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getMetadataInfo(result: any, metadataKeys?: string[]): SchemaEntityMetadata {
+    const entityMetadata: SchemaEntityMetadata = {};
+
+    if (metadataKeys) {
+        for (const key of metadataKeys) {
+            const attributePath = getAttributePath(key);
+            const value = attributePath.relativePath.length > 0
+                ? JSON.parse(result[attributePath.source])[attributePath.relativePath]
+                : result[attributePath.source];
+
+            switch (key) {
+                case 'content.entityname':
```

Comment:
nit: See if we can keep them as const list that can be expanded in future.

Reply:
Done